src/packet.rs

```diff
@@ -0,0 +1,49 @@
+//! Defines ITM packets and their possible contents.
+
+use heapless::Vec as HVec;
+use std::fmt::{self, Debug, Formatter};
+
+pub const MAX_PAYLOAD_SIZE: usize = 4;
+
+/// Represents a complete received packet.
+#[derive(Debug)]
+pub struct Packet {
```

Comment:
Same comment as the `Instrumentation` one. The user can construct `Packet`s with invalid header values, I think.

Reply:
Fixed. I think I also prefer accessor methods here, but because that allows us to change the internal representation freely.